libraries/AP_Param/AP_Param.cpp

```diff
@@ -1789,9 +1803,20 @@ bool AP_Param::load_defaults_file(const char *filename)
         return false;
     }
 
-    if (! read_param_defaults_file(filename)) {
-        return false;
+    saveptr = nullptr;
+    mutable_filename = strdup(filename);
+    if (mutable_filename == nullptr) {
+        AP_HAL::panic("AP_Param: Failed to allocate mutable string");
+    }
+    for (char *pname = strtok_r(mutable_filename, ",", &saveptr);
+         pname != nullptr;
+         pname = strtok_r(nullptr, ",", &saveptr)) {
+        if (!read_param_defaults_file(pname)) {
+            free(mutable_filename);
+            return false;
+        }
```

Comment:
what I was thinking was to use some kind of strv loop. We don't have a ready-to-use impl, so it would be more or less this:
https://gist.github.com/lucasdemarchi/1295579f4048f66dbd995864b5c031de

It doesn't allocate twice and has the more or less the same cost of iterating over the string.  Seeing the code now, I have mixed feelings. What do you think?